src/sentry/static/sentry/app/views/organizationDiscover/discover.jsx

```diff
@@ -44,6 +44,18 @@ export default class OrganizationDiscover extends React.Component {
     };
   }
 
+  componentWillReceiveProps(nextProps) {
+    const {queryBuilder, location: {search}} = nextProps;
+    if (search === '') {
```

Comment:
Is it possible that `search` here be any other falsey string? Can we early return here?